### DISCUSSION TITLE
[Node] Automatic grouping of @types/foo with foo?

### DISCUSSION THREAD

**Discussion by fgblomqvist** [renovatebot/renovate#10917] [Request Help]
Has there been any discussions/work on automatically grouping `@types/` packages with their corresponding package? Or is the `@types/` lag often big enough to where it is rare to update things at the same time?

E.g. the logic would follow what is said [here](https://github.com/DefinitelyTyped/DefinitelyTyped#how-do-definitely-typed-package-versions-relate-to-versions-of-the-corresponding-library) and use an algo similar to this:

Scenario: There is an update available to `foo` to version `x.y.z`
1. Check if `@types/foo` is also a dep
  a. If not, ok
2. If it is, check if there is an update to `@types/foo` to version `<=x.y.*`
  a. If not, ok
3. If there is, group them in one MR

There is also the scenario where Renovate finds an update to `@types/foo` first, but that can be similar.
Is this considered too much magic to be handled by Renovate?

**Comment by rarkins** [comment]
@fgblomqvist this is one of the oldest open issues we have: https://github.com/renovatebot/renovate/issues/519

I'm not sure if you saw but we do implicitly group `foo` with `@types/foo` already, IF `foo` wasn't *manually* grouped. We do this by putting `@types/foo` in e.g. `renovate/foo-2.x` instead of `renovate/types-foo-2.x`. So the challenge is "following" `foo` into whatever group it got put in.

BTW I dislike having manager-specific logic in our branch grouping code - which is 99% manager-agnostic - however this is definitely a most common use case which would benefit users. Another one - probably even harder - is figuring out peer dependencies and dynamically grouping those.
